sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -139,6 +140,13 @@ case class HiveTableScanExec(
   }
 
   protected override def doExecute(): RDD[InternalRow] = {
+    val locationPath = new Path(relation.catalogTable.location)
+    val fs = locationPath.getFileSystem(sparkSession.sessionState.newHadoopConf())
+
+    // if the table location is not exists, return an empty RDD
+    if (!fs.exists(locationPath)) {
+      return new EmptyRDD[InternalRow](sparkSession.sparkContext)
```

Comment:
It seems not reasonable to do this for partition table, because the real partition path maybe not under the location of the partition table, I moved this logic to `makeRDDForTable `